api/v1alpha1/virtualmachine_conversion.go

```diff
@@ -265,6 +265,12 @@ func convert_v1alpha1_NetworkInterface_To_v1alpha2_NetworkInterfaceSpec(
 	case "nsx-t":
 		out.Network.TypeMeta.APIVersion = "vmware.com/v1alpha1"
 		out.Network.TypeMeta.Kind = "VirtualNetwork"
+	case "nsx-t-subnet":
```

Comment:
Add these to the VM hub-spoke-hub test around https://github.com/vmware-tanzu/vm-operator/blob/af9c0ad4023ba8ff8e21e6691becfef2b8e2e7da/api/v1alpha1/virtualmachine_conversion_test.go#L83